src/coffee/directives/api/map.coffee

```diff
@@ -115,6 +116,11 @@ angular.module('uiGmapgoogle-maps.directives.api')
               scope.deferred.resolve _gMap
               resolveSpawned()
 
+            # run the ready event
+            if angular.isDefined scope.ready
+              scope.$applyAsync ->
```

Comment:
This is at the wrong indention level. This needs to be inside the idle callback, below `resolveSpawned()`.

Also the logic seems terse. Just do scope.`$parent.$evalAsync scope.ready if angular.isDefined scope.ready `